rmrk-token/tests/utils.rs

```diff
@@ -14,28 +14,624 @@ pub const BASE_ID: u64 = 3;
+        .expect("Failed to decode error");
+    if !error.contains(exp_error) {
+        println!("Received panic {:?}", error);
+        println!("Expected panic {:?}", exp_error);
+        panic!("");
+    }
+}
+
+pub trait RMRKToken {
+    fn rmrk(sys: &System, resource_hash: Option<[u8; 32]>) -> Program;
+    fn mint_to_root_owner(
+        &self,
+        user: u64,
+        root_owner: u64,
+        token_id: u64,
+        exp_error: Option<&str>,
+    );
+    fn mint_to_nft(
+        &self,
+        user: u64,
+        parent_id: u64,
+        parent_token_id: u64,
+        token_id: u64,
+        exp_error: Option<&str>,
+    );
+    fn burn(&self, token_id: u64, user: u64, exp_error: Option<&str>);
+    fn accept_child(
+        &self,
+        user: u64,
+        parent_token_id: u64,
+        child_contract_id: u64,
+        child_token_id: u64,
+        exp_error: Option<&str>,
     );
-    if resource_hash.is_some() {
-        println!("{:?}", res.decoded_log::<RMRKEvent>());
-        assert!(!res.log().is_empty());
-    } else {
-        assert!(res.log().is_empty());
+    fn reject_child(
+        &self,
+        user: u64,
+        parent_token_id: u64,
+        child_contract_id: u64,
+        child_token_id: u64,
+        exp_error: Option<&str>,
+    );
+    fn remove_child(
+        &self,
+        user: u64,
+        parent_token_id: u64,
+        child_contract_id: u64,
+        child_token_id: u64,
+        exp_error: Option<&str>,
+    );
+    fn approve(&self, user: u64, to: u64, token_id: u64);
+    fn transfer(&self, from: u64, to: u64, token_id: u64, exp_error: Option<&str>);
+    fn transfer_to_nft(
+        &self,
+        from: u64,
+        to: u64,
+        token_id: u64,
+        destination_id: u64,
+        exp_error: Option<&str>,
+    );
+    fn check_rmrk_owner(&self, token_id: u64, expected_token_id: Option<TokenId>, owner_id: u64);
+    fn check_balance(&self, user: u64, balance: u64);
+    fn check_pending_children(
+        &self,
+        parent_token_id: u64,
+        expected_pending_children: BTreeSet<(CollectionId, TokenId)>,
+    );
+    fn check_accepted_children(
+        &self,
+        parent_token_id: u64,
+        expected_accepted_children: BTreeSet<(CollectionId, TokenId)>,
+    );
+    fn check_root_owner(&self, token_id: u64, root_owner: u64);
+}
+
+impl RMRKToken for Program<'_> {
+    fn rmrk(sys: &System, resource_hash: Option<[u8; 32]>) -> Program {
+        let rmrk = Program::current(sys);
+        let res = rmrk.send(
+            USERS[0],
+            InitRMRK {
+                name: "RMRKToken".to_string(),
+                symbol: "RMRKSymbol".to_string(),
+                resource_hash,
+                resource_name: "ResourceName".to_string(),
+            },
+        );
+        if resource_hash.is_some() {
+            assert!(!res.log().is_empty());
+        } else {
+            assert!(res.log().is_empty());
+        }
```

Comment:
If I understand correctly, this is used to check a success of initialization, so you can use just the `main_failed()` method.
```suggestion
        assert!(!res.main_failed());
```